src/Commands/GenerateDomain.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\File;
+
+class GenerateDomain extends Command
+{
+    /**
+     * Domain Name.
+     *
+     * @var
+     */
+    public $name;
+
+    /**
+     * Directory Name.
+     *
+     * @var
+     */
+    public $directory;
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'make:domain {name : Domain Name}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a domain folder structure';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $this->name = ucfirst($this->argument('name'));
+
+        $this->generateDirectories();
```

Comment:
Still figuring out how to auto generate files specific for each folder.

Reply:
https://github.com/laravel/framework/blob/8.x/src/Illuminate/Foundation/Console/JobMakeCommand.php